coalib/results/TextPosition.py

```diff
@@ -47,6 +47,19 @@ def line(self):
     def column(self):
         return self._column
 
+    @property
+    def position(self):
+        if self._position is None:
+            # _position is calculated lazily. i.e. ONLY when position
+            # is asked (for the first time) If a user never tries to use
+            # `position` it is never computed.
+            if self._text is not None:
+                self._position = self.from_line_col(
```

Comment:
I'm a little confused here (sorry).
If we have this, why do we calculate `position` in `from_line_col` with the arg `create_object=True` ?
It would be better to let this property calculate it rather than do the calculation before hand right ?
(Basically, if a user does `from_line_col(text, line, col)` and never uses `position` we unnecessarily calculate it ...)


Maybe we can just change that to:
```
 +    def from_line_col(cls, text, line, column, create_object=True):
 +        """
 +        Calculates position in a file from line number and column.
 +        Uses \\n as the newline character. Position starts from 0.
 +
 +        :param text:   A tuple of lines in which position is to be calculated.
 +        :param line:   The line through which position is identified.
 +        :param column: Column through which position is idnetified.
 +        :create_object:     Boolean which determines whether to return
 +                            a TextPosition Object or a tuple.
 +        :return:            a TextPosition object or a tuple of the form
 +                            (line, column).
 +        """
 +        if create_object:
 +            return cls(line, column, text=text)
 +        try:
 +            position = sum(len(phrase) for phrase in text[:line-1])
 +            position += len(text[line-1][:column])-1
 +        except IndexError:
 +            raise ValueError("Position not found in text")
 +        return position   
```